packages/report-server/src/reportBuilder/output/functions/matrix/matrixBuilder.ts

```diff
@@ -58,20 +64,37 @@ export class MatrixBuilder {
     assignColumnSetToMatrixData(columnsInArray);
   }
 
+  private buildPeriod() {
+    const isString = (val: unknown): val is string => typeof val === 'string';
+    const periods = this.rows
+      .map(row => row.period)
+      .filter(isString)
+      .sort();
+
+    if (periods.length > 0) {
+      const earliestPeriod = periods[0];
+      const latestPeriod = periods[periods.length - 1];
+      this.metadata.period = {
+        ...this.metadata.period,
+        reportStart: earliestPeriod,
+        reportEnd: latestPeriod,
```

Comment:
I thought about overriding the `earliestAvailable` and `latestAvailable` values that we get from the intial fetch here, but then felt that maybe retaining both pieces of information could be valuable?

Reply:
I actually feel there are a few too many ways to refer to dates flying around the place, especially when you factor in `dataStartDate` and `dataEndDate` that these get turned into before going to the front end

What about bringing some consistency with something like:
- `earliestRawData`, `latestRawData` for what's returned from initial fetch (will require some refactor...)
- `earliestReportData`, `latestReportData` for what's calculated here
- `earliestDataDate`, `latestDataDate` for what is passed to the front end (doesn't sit inside a `period` object, so needs the extra `Date` context)

I quite like maintaining the divide between `start`/`end` as a directive or request from the client, vs. `earliest`/`latest` as information about the response from the server

Reply:
Ah good call, I hadn't considered the semantics of `start` vs `earliest` but it's a poignant distinction 👍 Happy to make these changes 😄